geth/node/manager.go

```diff
@@ -69,14 +77,108 @@ func (m *NodeManager) startNode(config *params.NodeConfig) (<-chan struct{}, err
 
 	m.initLog(config)
 
-	ethNode, err := MakeNode(config)
+	deliveryManager := new(notifications.DeliveryService)
+
+	ethNode, err := MakeNode(config, deliveryManager)
 	if err != nil {
 		return nil, err
 	}
 
+	m.deliveryService = deliveryManager
 	m.nodeStarted = make(chan struct{}, 1)
 
+	// Subscribe for message delivery status and log out with special key.
+	messageStateLoggingID := deliveryManager.Subscribe(func(state notifications.DeliveryState) {
```

Comment:
So big anonymous function.. It's hard to read. Do you really need anonymous function it there?

Reply:
Looks simpler now!) Do you really need anonymous function it there? I think it could be extracted to separated function.

Reply:
Refactoring that now ;)